src/arrays/ladderops.jl

```diff
@@ -0,0 +1,68 @@
+###########################
+# Ladder Operator Methods #
+########################### 
+    # n specifies which particle (a.k.a tensor product 
+    # factor) the operator acts on
+    raisematrix(lens, n=1) = laddermatrix(lens, RaiseOpFlag, n)
+    lowermatrix(lens, n=1) = laddermatrix(lens, LowerOpFlag, n)
+
+    raiseop(b::AbstractBasis, n=1) = QuArray(raisematrix(size(b), n), b, b)
+    raiseop(lens::Tuple, n=1) = raiseop(FiniteBasis(lens), n)
+    
+    lowerop(b::AbstractBasis, n=1) = QuArray(lowermatrix(size(b), n), b, b)
+    lowerop(lens::Tuple, n=1) = lowerop(FiniteBasis(lens), n)
+
+    ##########################
+    # Helper Functions/Types #
+    ##########################
+    abstract LadderOpFlag
+    abstract RaiseOpFlag <: LadderOpFlag
+    abstract LowerOpFlag <: LadderOpFlag    
+
+    ladder_inds(n, ::Type{RaiseOpFlag}) = ([1:n-1], [2:n])
+    ladder_inds(n, ::Type{LowerOpFlag}) = ([2:n], [1:n-1])
+    laddercoeffs(n) = complex(sqrt(linspace(1, n, n)))
```

Comment:
It's actually slightly more performant to do the conversion here than at the time of `setindex!` when filling the array in `fill_op_arr!` (this is of course assuming that we want our "default" element type to be `Complex128`, which I believe we do). I think the performance boost is due to it being easier for the compiler to optimize a map versus the mutation that would otherwise have to occur in the `for` loop, though I'm not entirely sure.